modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/BaseResolverFactoryBean.java

```diff
@@ -75,16 +74,4 @@ public void initialise() throws InitialisationException
     {
         LifecycleUtils.initialiseIfNeeded(valueResolver, muleContext);
     }
-
-    /**
-     * Propagates the disposal event to the
-     * underlying {@link #valueResolver} if it implements
-     * the {@link Disposable} interface
-     */
-    @Override
-    public void dispose()
```

Comment:
this was a workaround to solve the issues that motiveated me to engage on this refactor. The initialise was not removed because at that point the resolved value is not reachable by the lifecycle manager.

What I fixed by removing this dispose() call is the ValueResolver getting the dispose() twice. However, this whole part of the ext-api is going to be rewritten now that I have this cool new features so I wouldn't really invest time on this for this PR